mkosi/gentoo.py

```diff
@@ -0,0 +1,549 @@
+            MkosiPrinter.info(PORTAGE_INSTALL_INSTRUCTIONS)
+            raise MkosiException(e)
+
+        return dict(profile_path=PROFILE_PATH,
+                    custom_profile_path=CUSTOM_PROFILE_PATH,
+                    ebuild_sh_env_dir=EBUILD_SH_ENV_DIR,
+                    portage_cfg_dir=USER_CONFIG_PATH)
+
+    def __init__(
+        self,
+        args: CommandLineArguments,
+        root: Path,
+        do_run_build_script: bool,
+    ) -> None:
+
+        ret = self.try_import_portage()
+
+        from portage.package.ebuild.config import config  # type: ignore
+
+        self.portage_cfg = config(config_root=str(root), target_root=str(root),
+                                  sysroot=str(root), eprefix=None)
+
+        PORTAGE_MISCONFIGURED_MSG = "You have portage(5) installed but it's probably missing defaults, bailing out"
+        # we check for PORTDIR, but we could check for any other one
+        if self.portage_cfg['PORTDIR'] is None:
+            die(PORTAGE_MISCONFIGURED_MSG)
+
+        if "build-script" in ARG_DEBUG:
+            for k, v in self.portage_cfg.items():
+                print(f"{k} = {v}")
+
+        self.profile_path = root / ret["profile_path"]
+        self.custom_profile_path = root / ret["custom_profile_path"]
+        self.ebuild_sh_env_dir = root / ret["ebuild_sh_env_dir"]
+        self.portage_cfg_dir = root / ret["portage_cfg_dir"]
+
+        self.portage_cfg_dir.mkdir(parents=True, exist_ok=True)
+
+        self.DEFAULT_NSPAWN_PARAMS = [
+            "--capability=CAP_SYS_ADMIN,CAP_MKNOD",
+            "--tmpfs=/sys",
+            f"--bind={self.portage_cfg['PORTDIR']}",
+            f"--bind={self.portage_cfg['DISTDIR']}",
+            f"--bind={self.portage_cfg['PKGDIR']}",
+        ]
+
+        jobs = multiprocessing.cpu_count()
+        self.emerge_default_opts = [
+            "--buildpkg=y",
+            "--usepkg=y",
+            "--keep-going=y",
+            f"--jobs={jobs}",
+            f"--load-average={jobs-1}",
+            "--nospinner",
+        ]
+        if "build-script" in ARG_DEBUG:
+            self.emerge_default_opts += ["--verbose", "--quiet=n", "--quiet-fail=n"]
+        else:
+            self.emerge_default_opts += ["--quiet-build", "--quiet"]
+
+        global kimg_path
+        self.arch, kimg_path = self.ARCHITECTURES[args.architecture or "x86_64"]
+
+        # GENTOO_UPSTREAM : we only support systemd profiles! and only the no-multilib flaivour , for now;
+        # GENTOO_UPSTREAM : wait for fix upstream: https://bugs.gentoo.org/792081
+        # GENTOO_TODO     : add args.profile switch for this? (multilib vs. nomultilib)
+        # GENTOO_DONTMOVE : could be done inside set_profile, however stage3_fetch() will be needing this if we want to allow users to pick profile
+        self.arch_profile = Path(f"profiles/default/linux/{self.arch}/{args.release}/no-multilib/systemd")
+
+        self.pkgs_sys = ["@world"]
+
+        self.pkgs_fs = ["sys-fs/dosfstools"]
```

Comment:
Could you add some comments what `pkgs_sys` and `pkgs_fs` are or make the names more descriptive? Sorry if I missed that elsewhere, but it would greatly help to understand what's going on if this needs to be maintained when you're not there. :)